src/gui/src/tattoowidget.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * This file is part of the QPxTool project.
+ * Copyright (C) 2009 Gennady "ShultZ" Kozlov <qpxtool@mail.ru>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * See the file "COPYING" for the exact licensing terms.
+ *
+ */
+
+#include <cmath>
+
+#include <QDebug>
+#include <QPainter>
+
+#include "tattoowidget.hpp"
+
+TattooWidget::TattooWidget(QWidget *p) : QWidget(p) {
+  r0mm = F1TATTOOR / 3;
+  r1mm = F1TATTOOR;
+}
+
+TattooWidget::~TattooWidget() {}
+
+void TattooWidget::setRadius(int ir0, int ir1) {
+  r0mm = ir0;
+  r1mm = ir1;
+  convertImage();
+  update();
+}
+
+void TattooWidget::setImage(QImage &iimg) {
+  timg = iimg;
+  convertImage();
+  update();
+}
+
+#define CONVERT_DIRECT
+
+void TattooWidget::convertImage() {
+  x0 = width() / 2;
+  y0 = height() / 2;
+  if (width() < height()) {
+    r1p = x0 - 4;
+  } else {
+    r1p = y0 - 4;
+  }
+
+  r0p = (int)((float)r1p * r0mm / r1mm);
+
+  dimg = QImage(r1p << 1, r1p << 1, QImage::Format_ARGB32);
+  dimg.fill(0xFFFFFFFF);
+
+  float ang, rad;
```

Comment:
Codacy found an issue: [The scope of the variable 'ang' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)